guarddog/analyzer/metadata/npm/direct_url_dependency.py

```diff
@@ -0,0 +1,64 @@
+""" Direct URL Dependency Detector
+
+Detects if a package depends on direct URL dependencies
+"""
+from typing import Optional
+import re
+
+from guarddog.analyzer.metadata.detector import Detector
+
+from urllib.parse import urlparse
+
+
+github_project_pattern = re.compile(r"^([\w\-\.]+)/([\w\-\.]+)")
+
+
+class NPMDirectURLDependencyDetector(Detector):
+    """This heuristic detects packages with direct URL dependencies.
+    Dependencies fetched this way are not immutable and can be used to inject untrusted code
+    or reduce the likelihood of a reproducible install."""
+
+    def __init__(self):
+        super().__init__(
+            name="direct_url_dependency",
+            description="Identify packages with direct URL dependencies",
```

Comment:
can we give slightly more information here so it ends up in the autogenerated docs (you can generate it with `make docs`)

Reply:
Just did in 5a95438f5a337e1f7a8ad1bfa43ff8b5a68c7cd7.